### PR TITLE
Task/dr 332 lambda deploy infra

### PR DESCRIPTION
**Pull request by Pira-nhs**
## Description

The following AWS Lambdas are spun up via terraform and provisioned via a CI/CD pipeline. The AWS Lambda is functionals is able to interact with the associated existing Dynamodb tables. 

- Organisations 
- Organisation_Affliation
- Locations
- Healthcare_Services 
- Questionnaire 
- Questionnaire Response

There will be additional work required later to connect the API gateways & microservices with the Lambdas, code is submitted with this in consideration.

## Context

This change is required in order to deploy AWS Lambda in order to make contact and interact with the existing Dynamo DB tables. 

## Type of changes

- [ ] Refactoring (non-breaking change)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [x] I have followed the code style of the project
- [x] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [x] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [x] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

File: infrastructure/modules/lambda/output.tf [hunk 1]
```diff
@@ -0,0 +1 @@
+#
```
Comment (timrickwood) [inline infrastructure/modules/lambda/output.tf:1]: Do we need an empty file ? Or should we add it when we actually use it ?

File: infrastructure/stacks/capacity_management/lambda.tf [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+
+#    allowed_triggers = {
+#        AllowExecutionFromAPIGateway = {
+#        service    = "apigateway"
+#        source_arn = "${aws_api_gateway_rest_api.DoS_REST.execution_arn}/*/*"
+#        }
+#    }
+#    }
+
+#    module "live-alias-organisations-data-manager" {
```
Comment (timrickwood) [inline infrastructure/stacks/capacity_management/lambda.tf:48]: Dead code we can remove ?

File: infrastructure/stacks/capacity_management/lambda.tf [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+
+#    allowed_triggers = {
+#        AllowExecutionFromAPIGateway = {
+#        service    = "apigateway"
+#        source_arn = "${aws_api_gateway_rest_api.DoS_REST.execution_arn}/*/*"
+#        }
+#    }
+#    }
+
+#    module "live-alias-organisation-affiliation-data-manager" {
```
Comment (timrickwood) [inline infrastructure/stacks/capacity_management/lambda.tf:101]: Dead code we can remove ?

File: infrastructure/stacks/capacity_management/lambda.tf [hunk 4]
```diff
@@ -0,0 +1,319 @@
...
+#    allowed_triggers = {
+#        AllowExecutionFromAPIGateway = {
+#        service    = "apigateway"
+#        source_arn = "${aws_api_gateway_rest_api.DoS_REST.execution_arn}/*/*"
+#        }
+#    }
+#    }
+
+#    module "live-alias-healthcare-services-data-manager" {
+#    source = "terraform-aws-modules/lambda/aws//modules/alias"
```
Comment (timrickwood) [inline infrastructure/stacks/capacity_management/lambda.tf:208]: See similar comments?

File: infrastructure/modules/lambda/output.tf (hunk 1 above)
Comment (Pira-nhs) [reply infrastructure/modules/lambda/output.tf:1]: There is no use for it currently was thinking for future use but can remove and add later when needed.

File: infrastructure/stacks/capacity_management/lambda.tf (hunk 2 above)
Comment (Pira-nhs) [reply infrastructure/stacks/capacity_management/lambda.tf:48]: Needed as part of the work coming up in the next sprint

File: infrastructure/stacks/capacity_management/lambda.tf (hunk 4 above)
Comment (Pira-nhs) [reply infrastructure/stacks/capacity_management/lambda.tf:208]: Needed as part of the work coming up in the next sprint ..same as above

**Review by timrickwood — APPROVED** [review]

**Review by gdcNHS — APPROVED** [review]

**Pull request merged**